### DISCUSSION TITLE
Barra de colores

### DISCUSSION THREAD

**Discussion by MaCuur** [grupoioa/gis_agm_series#26] [General]
Hola, abro esta discusión para elaborar una descripción más a fondo sobre el conflicto en el que nos  enredamos hoy.
@cuausv 
@ma-robles 
@AbrilIchel

**Comment by AbrilIchel** [comment]
Para aclarar el tema voy a copiar la explicación que les di en whats, para comenzar la discusión a partir de ahí:

En algún momento se hablo de que los colores de la barra tenían que coincidir con la imagen, (yo entendí que se tenían que hacer de una manera independiente Imagen-Barra), la idea que pensé para que no batallarán tanteandole a los colores, era hacer un PNG en MatLab donde los colores estuvieran directamente relacionados a la barra.

Puntualizando que se esta hablando únicamente de las imágenes que se van a utilizar para la despliegue y descarga( dicese  dinámicas).
Cual fue la idea que tuviste @ma-robles ???

**Reply by ma-robles** [reply]
Primero, aclarando, la imagen y la barra las puede generar ncWMS y coinciden una con otra, siempre y cuando se soliciten con los mismos parámetros. Sin embargo está barra que genera está medio chafa, por lo que se pensó en hacer una barra de manera independiente, el detalle sería verificar que los colores imagen- barra coincidan. Por cierto, en ncWMS, el problema sería solicitar la barra de manera que coincida exactamente con las barras que ya existen.

**Reply by AbrilIchel** [reply]
Buscando información, leí en alguna parte que las figuras que están en OWGIS (Homologas a las que se quieren hacer para esta sección), están en formato KML. Solo es un comentario para ver si es útil esta información, y si es posible explorar esa opción. Estas figuras se pueden exportan con ese formato en MatLab, conserva las características de formato y geográficas.

**Reply by ma-robles** [reply]
@AbrilIchel no creo que estén en KML, mas bien es posible que el owgis te genere un KML (aunque mas bien los debe hacer algún servicio WMS). pero me parece que no tienen mucho que ver

**Comment by cuausv** [comment]
En mapas dinámicos no en estático. Ok entiendo

**Reply by ma-robles** [reply]
así es @cuausv, la idea es para dinámicos, pero si tienes alguna idea de donde podamos hacer las series a partir de los estáticos también es bienvenida

**Comment by ma-robles** [comment]
Bueno, lo que estaba pensando es que... En realidad no necesitamos una barra de colores, ni su imagen correspondiente.
Nosotros vamos a generar una serie de tiempo, para un punto en específico. Sólo necesitamos el punto, la variable y el tipo. En cambio, la imagen que se proyecta en el mapa es el estado para todo el dominio de una variable y tipo en un momento específico (tiempo)

**Reply by MaCuur** [reply]
pero se necesita como referencia para, al menos, hacer una simple inspección de la imagen que se está observando.

**Reply by ma-robles** [reply]
No es muy útil (creo) dado que el mapa es para un momento dado, digamos enero para un estadístico mensual, y la serie de tiempo es para todo el intervalo (todo el año de para el ejemplo)

**Comment by ma-robles** [comment]
Es decir, en cuestión del mapa, necesitamos algo [así]( https://www.latlong.net/) pero para el golfo de México. Con esta página o mapa, el usuario indica un punto y después (o ahí mismo) se le solicita la variable y tipo de estadístico. Con esta info se genera la serie de tiempo

**Reply by AbrilIchel** [reply]
Si en eso tienes razón, estrictamente no necesitamos una imagen, lo principal es que descargue y despliegue, digamos que la parte de imagen y colores le daría calidad, al final de cuentas también se tiene que pensar en lo bonito, tendría que ser una imagen representativa promedio de todo o el mínimo máximo de todo . Eso lo podemos dejar como algo secundario, ya que lo prioritario este resuelto, o ustedes qué opinan?

**Reply by ma-robles** [reply]
Exacto @AbrilIchel, pensando en lo que dijo @MaCuur, en que se necesita una imagen de referencia, esta podría ser una capa de batimetría o algo así. Tal vez ella tenga mejores opciones :)

**Reply by MaCuur** [reply]
> Exacto @AbrilIchel, pensando en lo que dijo @MaCuur, en que se necesita una imagen de referencia, esta podría ser una capa de batimetría o algo así. Tal vez ella tenga mejores opciones :)

la barra de colores sería por simple cortesía para quién ve la imagen. Pero igual  se me ocurre, en vez de la batimetría, que sean isotermas, para cubrir ese detallito que mencionaba de tener la noción de la imagen que se está observando.

Sobre que es solamente el hecho de que será solamente un instante lo que se está mostrando, se detallaría en el manual de uso.

**Reply by ma-robles** [reply]
@MaCuur pero... ¿isotermas de que momento? ¿mensual? ¿anual? ¿promedio? ¿mínima?¿ máxima?y pues... en realidad no es muy útil observar  una variable, que probablemente no corresponda al momento/tipo que te interesa. Creo que sería más útil una referencia o ayuda para ubicar tu punto adecuadamente. Por ejemplo, si hay un derrame en alguna plataforma petrolera,  pues será de mucha ayuda  tener indicada ahí la plataforma, sólo la selecciono de la manera que se decida (con un click, selección de menú, coordenadas, etc) y veo la variable que me interese, viento, por ejemplo para el mes en que estamos

**Reply by MaCuur** [reply]
> @MaCuur pero... ¿isotermas de que momento? ¿mensual? ¿anual? ¿promedio? ¿mínima?¿ máxima?y pues... en realidad no es muy útil observar una variable, que probablemente no corresponda al momento/tipo que te interesa. Creo que sería más útil una referencia o ayuda para ubicar tu punto adecuadamente. Por ejemplo, si hay un derrame en alguna plataforma petrolera, pues será de mucha ayuda tener indicada ahí la plataforma, sólo la selecciono de la manera que se decida (con un click, selección de menú, coordenadas, etc) y veo la variable que me interese, viento, por ejemplo para el mes en que estamos

de ese instante de la variable desplegada, pero ya mejor así como la pensaron está bien, ahora que sepan cuales son los puntos que le interesan a pemex, igual y pueden ser esos puntos los que se muestren ahí

jeje,

**Reply by ma-robles** [reply]
me parece que mostrar esas isotermas que mencionas @MaCuur va mejor si quisieramos hacer la animación del mapa completo, pero eso ya se hace en OWGIS y no es nuestro objetivo por el momento. Claro, podríamos ir pensando en abarca el despliegue completo, pero una vez que tengamos cubierto nuestro objetivo.

**Reply by MaCuur** [reply]
sip :)

**Comment by ma-robles** [comment]
Entonces, concluyendo: no necesitamos barras de colores, al menos no como lo habíamos pensado. En su lugar busquemos algo más para hacer mas "amigable" el mapa. Abro discusión nueva para las propuestas [https://github.com/grupoioa/gis_agm_series/discussions/27](https://github.com/grupoioa/gis_agm_series/discussions/27). Cierro esta discusión
